snap_plugin/v1/plugin.py

```diff
@@ -436,6 +464,26 @@ def _parse_args(self):
                   self._is_shutting_down),
             kwargs={"timeout": self._get_ping_timeout_duration()})
 
+        # check if the secure flag has been added
+        self.tls_enabled = self._args.tls_enabled
```

Comment:
I think there's no need at all to create `self.tls_enabled` as `self._args` is accessible everywhere and persistent after parsing, so there's no need to do additional storing.